### ISSUE TITLE
Power question

### ISSUE THREAD

**Issue by shift838** [walhi/arduino_eprom27_programmer#11]
Can this unit only be powered via the Nano, or does it require external power to program the eproms?  If using external power, what voltage adapter is it requiring, 24v? or what?

**Comment by walhi** [comment]
To flash the chip, high voltage is required. It can be obtained from the USB from the Arduino itself using a step-up converter or supplied from a laboratory power supply. What voltage exactly depends on the chip, this should be clarified in the datasheet.

**Comment by shift838** [comment]
i was just asking if you were using a 24v power adapter with the barrel connector I mean.

**Comment by walhi** [comment]
No, I plug a lab power supply into it.

**Comment by shift838** [comment]
but would a 24v power adapter work fine then?  and if so, is the center pole 24v or ground?

**Comment by walhi** [comment]
Yes, the central pin is a power plus. but let me remind you once again that you need to check the programming voltage of the microcircuit.

**Comment by shift838** [comment]
assuming the lab power supply you are hooking up to the X1 header connector?  If I am understanding the schematic correctly the MT3608 stepup DC module takes the 5v from the arduino and will step it up to the required voltage?  but I think you will need to configure that voltage of course based on the programming requirements.  Never worked with this stepup module so i'm assuming the module has an variable knob on it to get the required voltage?

So really it has 3 different ways it can get the power required.  Via Lab power supply, DC adapter or via the arduino power provided all through the step up converter?

**Comment by walhi** [comment]
Yes, you are right. There is a trimmer resistor on this step-up converter. For the convenience of the user, the firmware has a voltage measurement function built into it.

**Comment by shift838** [comment]
If I'm interpreting your .ino file correctly, it looks like it is setting the programming voltage based on the selected chip via the step-up module, so I would not have to adjust it.  If so, there is no need for me to install the barrel connector just use the power provided by the arduino or a lab power supply.  Am I correct in assuming that?
